controllers/vm_migration_controller.go

```diff
@@ -0,0 +1,193 @@
+	"github.com/j-keck/arping"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	virtv1 "kubevirt.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// VirtualMachineReconciler makes sure the VM has a correct mac address on the switch
+type VirtualMachineReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachineinstances,verbs=get;list;watch
+
+func (r *VirtualMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	log.Log.Info(fmt.Sprintf("VirtualMachine: Reconcile Virtual Machine %+v", req))
+
+	virtualMachineInstance := &virtv1.VirtualMachineInstance{}
+	err := r.Get(ctx, req.NamespacedName, virtualMachineInstance)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	virtualMachineNetworks := virtualMachineInstance.Spec.Networks
+
+	for i := range virtualMachineNetworks {
+		multusNetwork := virtualMachineNetworks[i].NetworkSource.Multus
+		// network type can be Multus or Pod. We will work only with Multus
+		if multusNetwork == nil {
+			continue
+		}
+
+		namespacedNetworkName := getNamespacedNetworkName(multusNetwork.NetworkName)
+
+		network := &networkv1alpha1.Network{}
+		err := r.Get(ctx, namespacedNetworkName, network)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		networkIpMasq := network.Spec.IpMasq
+		// send a garp request only if IP masquerading is enabled
+		if !networkIpMasq.Enabled {
+			continue
+		}
+
+		// arping -A -i <interface-name> -S 169.254.1.1 169.254.1.1
+		interfaceName := networkIpMasq.Bridge
+		log.Log.Info(
+			fmt.Sprintf(
+				"VirtualMachine: Sending a garp request to IP %s on interface %s for VM %s",
+				virtualIpaddress, interfaceName, req.Name),
+		)
+		err = arping.GratuitousArpOverIfaceByName(net.ParseIP(virtualIpaddress), interfaceName)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// Given a network name of the KubeVirt VM object,
+// construct a namespaced name that can be used in API requests
+func getNamespacedNetworkName(networkName string) types.NamespacedName {
+	var namespacedNetworkName types.NamespacedName
+
+	networkNameParts := strings.Split(networkName, "/")
+	// this means the network is provided with the namespace
+	// default/network-name
+	if len(networkNameParts) == 2 {
+		namespacedNetworkName.Namespace = networkNameParts[0]
+		namespacedNetworkName.Name = networkNameParts[1]
+	} else {
+		namespacedNetworkName.Name = networkNameParts[0]
+	}
+
+	return namespacedNetworkName
+}
+
+// check if the migration status of the VMI object indicates that the migration was successful
+func migrationSuccessful(vmi virtv1.VirtualMachineInstance) bool {
```

Comment:
What about as follows?
```go
func migrationSuccessful(vmi virtv1.VirtualMachineInstance) bool {
  return vmi.Status.MigrationState != nil && vmi.Status.MigrationState.Completed && !vmi.Status.MigrationState.Failed
}
```

Reply:
Looks very pythonic, I like it :smile: 
However, I tried to make it as readable as possible and this version looks long.
I updated the function to preserve readability but also decrease the number of lines, wdyt?